contracts/car-races/io/src/lib.rs

```diff
@@ -0,0 +1,377 @@
+//! Data types for the contract input/output.
+
+#![no_std]
+
+use gmeta::{In, InOut, Metadata};
+use gstd::{collections::BTreeMap, prelude::*, ActorId, MessageId};
+// prices for acceleration and shuffles are constants for simple implementation
+pub const ACCELERATION_COST: u32 = 10;
+pub const SHELL_COST: u32 = 20;
+
+pub const DEFAULT_ACC_AMOUNT: u32 = 20;
+pub const DEFAULT_SHELL_AMOUNT: u32 = 25;
+
+pub const MAX_ACC_AMOUNT: u32 = 40;
+pub const MAX_SHELL_AMOUNT: u32 = 40;
+pub const MAX_DISTANCE: u32 = 3_242;
+pub const TIME: u32 = 1;
+
+pub const DEFAULT_SPEED: u32 = 100;
+
+pub const GAS_FOR_STRATEGY: u64 = 20_000_000_000;
+pub const GAS_FOR_ROUND: u64 = 150_000_000_000;
+pub const RESERVATION_AMOUNT: u64 = 240_000_000_000;
+pub const RESERVATION_TIME: u32 = 86_400;
+pub const GAS_MIN_AMOUNT: u64 = 30_000_000_000;
+
+/// Time deadline for player turn(30_000ms).
+pub const TURN_DEADLINE_MS: u64 = 30_000;
+
+pub const MIN_SPEED: u32 = 10;
+
+/// Time after which the game instance must be removed
+/// 1 block = 3s (1 minutes)
+pub const TIME_INTERVAL: u32 = 20;
+
+/// Gas for deleting the game instance
+pub const GAS_TO_REMOVE_GAME: u64 = 20_000_000_000;
+
+/// Contract metadata. This is the contract's interface description.
+///
+/// It defines the types of messages that can be sent to the contract.
+pub struct ContractMetadata;
+
+impl Metadata for ContractMetadata {
+    /// Init message type.
+    ///
+    /// Describes incoming/outgoing types for the `init()` function.
+    ///
+    /// The [`GameInit`] type is passed for initial smart-contract data(i.e config..) if any.
+    type Init = In<GameInit>;
+    /// Handle message type.
+    ///
+    /// Describes incoming/outgoing types for the `handle()` function.
+    ///
+    /// We use the [`GameAction`] type for incoming and [`GameReply`] for outgoing
+    /// messages.
+    type Handle = InOut<GameAction, GameReply>;
+    /// Asynchronous handle message type.
+    ///
+    /// Describes incoming/outgoing types for the `main()` function in case of
+    /// asynchronous interaction.
+    ///
+    /// The unit tuple is used as we don't use asynchronous interaction in this
+    /// contract.
+    type Others = InOut<(), RoundInfo>;
+    /// Reply message type.
+    ///
+    /// Describes incoming/outgoing types of messages performed using the
+    /// `handle_reply()` function.
+    ///
+    /// The unit tuple is used as we don't process any replies in this contract.
+    type Reply = ();
+    /// Signal message type.
+    ///
+    /// Describes only the outgoing type from the program while processing the
+    /// system signal.
+    ///
+    /// The unit tuple is used as we don't process any signals in this contract.
+    type Signal = ();
+
+    type State = InOut<StateQuery, StateReply>;
+}
+
+#[derive(Encode, Decode, TypeInfo)]
+#[codec(crate = gstd::codec)]
+#[scale_info(crate = gstd::scale_info)]
```

Comment:
Too many types here, and these attributes look dirty. Maybe it's worth to import them from `codec` and `scale_info`.